### PR TITLE
added slicing tests

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

File: sup3r/preprocessing/data_handling.py [hunk 1]
```diff
@@ -561,10 +561,8 @@ def __init__(self, file_paths, features, target=None, shape=None,
             bad_shape = (sample_shape[0] > self.grid_shape[0]
                          and sample_shape[1] > self.grid_shape[1])
             if bad_shape:
-                msg = (f'spatial_sample_shape {sample_shape[:2]} is larger '
-                       f'than the raster size {self.grid_shape}')
-                logger.warning(msg)
-                warnings.warn(msg)
+                logger.debug(f'spatial_sample_shape {sample_shape[:2]} is '
```
Comment (grantbuster) [inline sup3r/preprocessing/data_handling.py:564]: @bnb32  i reduced the severity of these warnings, they're really just in the wrong place and generate a lot of noise. Thoughts?

Comment (bnb32) [reply sup3r/preprocessing/data_handling.py:564]: Yeah totally agree with this

File: sup3r/pipeline/forward_pass.py [hunk 2]
```diff
@@ -410,6 +410,26 @@ def get_hr_slices(slices, enhancement, step=None):
...
+    @property
+    def chunk_lookup(self):
+        """Get a 3D array with shape
+        (n_spatial_1_chunks, n_spatial_2_chunks, n_temporal_chunks)
+        where each value is the chunk index."""
+        n_s1 = len(self.s1_lr_slices)
+        n_s2 = len(self.s2_lr_slices)
+        n_t = self.n_temporal_chunks
+        lookup = np.arange(self.n_chunks).reshape((n_t, n_s1, n_s2))
+        lookup = np.transpose(lookup, axes=(1, 2, 0))
```
Comment (bnb32) [inline sup3r/pipeline/forward_pass.py:422]: How fast is the reshape? Do we want to only create this lookup array once?

**Review by bnb32 — APPROVED** [review]
Looks good. I'll make those small changes I mentioned.

**Review by bnb32 — APPROVED** [review]
Looks good. I'll make those small changes I mentioned.

File: sup3r/pipeline/forward_pass.py (hunk 2 above)
Comment (grantbuster) [reply sup3r/pipeline/forward_pass.py:422]: oh super fast, it's just the low res data shape. I wouldnt worry about it.

Comment (bnb32) [reply sup3r/pipeline/forward_pass.py:422]: Ok cool

**Pull request merged**